File: CivitasJava/civitas/SorpresaIrCasilla.java
package civitas;
import java.util.ArrayList;

public class SorpresaIrCasilla extends Sorpresa {
    private int valor;
    private Tablero tablero;
    
    SorpresaIrCasilla(Tablero _tablero, int _valor, String _texto){
        super(_texto);
        tablero = _tablero;
        valor = _valor;
    }
    
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) throws Exception{
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            int casillaActual = todos.get(actual).getNumCasillaActual();
            int tirada = tablero.calcularTirada(casillaActual, valor);
            int nuevaPos = tablero.nuevaPosicion(casillaActual, tirada);
            todos.get(actual).moverACasilla(nuevaPos);
            tablero.getCasilla(nuevaPos).recibeJugador(actual, todos);
        }
    }
}

File: CivitasRuby/sorpresa_ir_casilla.rb
require_relative 'sorpresa'
require_relative 'tablero'

module Civitas
class SorpresaIrCasilla < Sorpresa
  
  def initialize(tablero, valor, texto)
    super(texto)
    @tablero = tablero
    @valor = valor
  end
  
  def aplicar_a_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      casilla_actual = todos[actual].numCasillaActual
      tirada = @tablero.calcular_tirada(casilla_actual, @valor)
      nueva_pos = @tablero.nueva_posicion(casilla_actual, tirada)
      todos[actual].mover_a_casilla(nueva_pos)
      casilla = @tablero.get_casilla(nueva_pos)
      casilla.recibe_jugador(actual, todos)
    end
  end
  
  public_class_method :new
end
end
